### PR TITLE
Create cart

### PR DESCRIPTION
**Pull request by Ru-T**
## What Changed?
* Created pending_purchases table 
* Added index view for user's cart showing only their own pending purchases
* Added create action for add to cart, defaulting book, price_at_purchase, and quantity at creation
* Added destroy action for remove from cart
* Added edit action and view to update the quantity from the cart
* Passed integration suite for quantity adjustment

## Why? 
* To allow users to add their books to a shopping cart before purchasing them

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -3,4 +3,21 @@ class User < ActiveRecord::Base
...
+                              quantity: 1
+                             )
+  end
+
+  def remove_from_cart(book)
+    pending_purchases.find_by_book_id(book.id).destroy
+  end
+
+  def in_cart?(book)
+    pending_purchases.any? { |h| h[:book_id] == book.id }
```
Comment (jahammo2) [inline app/models/user.rb]: What's `h`?

Comment (Ru-T) [inline app/models/user.rb]: hash. I can call it as such. Unless you know of a better way to do this?

File: app/models/user.rb [hunk 2]
```diff
@@ -3,4 +3,21 @@ class User < ActiveRecord::Base
...
   devise :database_authenticatable, :registerable, :confirmable,
          :recoverable, :rememberable, :trackable, :validatable
+
+  has_many :pending_purchases
+
+  def add_to_cart(book)
+    pending_purchases.create!(book_id: book.id,
+                              price_at_purchase: book.price,
+                              quantity: 1
+                             )
```
Comment (jahammo2) [inline app/models/user.rb]: This syntax is kind of strange. It'd probably be better like this:
```
pending_purchases.create!(
  book_id: book.id,
  price_at_purchase: book.price,
  quantity: 1
)
 ```

Comment (Ru-T) [inline app/models/user.rb]: Totally. Looked weird to me too. Fixed it.

File: app/views/pending_purchases/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+  <% @pending_purchases.each do |purchase| %>
+    <p> <%= purchase.user.email %> </p>
+    <p> <%= purchase.book.title %> </p>
+    <p> <%= purchase.quantity %>
+        <%= link_to 'Edit Quantity', edit_pending_purchase_path(purchase), id: "edit_quantity" %>
+    </p>
+    <p> <%= purchase.price_at_purchase %> </p>
+    <%= form_for(current_user.pending_purchases.find_by_book_id(purchase.book.id),
+                  method: :delete) do |f| %>
```
Comment (jahammo2) [inline app/views/pending_purchases/index.html.erb]: Same here. I get that Hound was probably yelling at you. Maybe like this:
```
<%= form_for(
  current_user.pending_purchases.find_by_book_id(purchase.book.id),
  method: :delete
) do |f| %>
```

That's just a suggestion. If you think it makes your code less readable. Don't do it.

File: db/seeds.rb [hunk 4]
```diff
@@ -22,3 +22,11 @@
 AdminUser.create!(email: 'admin@example.com',
                   password: 'password',
                   password_confirmation: 'password')
+
+20.times do
+  PendingPurchase.create!(user_id: rand(1...21),
+                          book_id: rand(1...100),
+                          price_at_purchase: rand(1...100),
+                          quantity: 1
+                         )
```
Comment (jahammo2) [inline db/seeds.rb]: Same.
```
PendingPurchase.create!(
  user_id: rand(1...21),
  book_id: rand(1...100),
  price_at_purchase: rand(1...100),
  quantity: 1
)
```

File: app/models/user.rb (hunk 1 above)
Comment (jahammo2) [inline app/models/user.rb]: I think it would probably be better as `pending_purchase`. Just because that's very clear. Brevity isn't necessary with code.

**Comment by jahammo2** [comment]
This was really well tested! I have some syntax issues and suggestions but it's good code. :shipit:

File: app/models/user.rb (hunk 1 above)
Comment (Ru-T) [inline app/models/user.rb]: :+1:

File: app/views/pending_purchases/index.html.erb (hunk 3 above)
Comment (Ru-T) [inline app/views/pending_purchases/index.html.erb]: haha cool.

File: db/seeds.rb (hunk 4 above)
Comment (Ru-T) [inline db/seeds.rb]: fixed.

**Pull request merged**